src/__tests__/Bugsnag.test.js

```diff
@@ -337,3 +337,138 @@ test('{enable|disable}ConsoleBreadCrumbs(): gracefully handles serialization edg
   // switched off for the next test ¯\_(ツ)_/¯
   c.disableConsoleBreadCrumbs()
 })
+
+
+test('typedMap converts arrays, objects, strings and numbers', () => {
+  const { typedMap } = require('../Bugsnag')
+  expect(typedMap({
+    arr: [ 0, 1, 2 ],
+    obj: {
+      a: 'a',
+      b: 'b',
+      c: 'c'
+    },
+    x: 10,
+    y: 'str'
+  })).toEqual({
+    arr: {
+      type: 'map',
+      value: {
+        '0': { type: 'number', value: 0 },
+        '1': { type: 'number', value: 1 },
+        '2': { type: 'number', value: 2 }
+      }
+    },
+    obj: {
+      type: 'map',
+      value: {
+        a: { type: 'string', value: 'a' },
+        b: { type: 'string', value: 'b' },
+        c: { type: 'string', value: 'c' }
+      }
+    },
+    x: { type: 'number', value: 10 },
+    y: { type: 'string', value: 'str' }
+  })
+})
+
+test('typedMap handles circular structures', () => {
+  const { typedMap } = require('../Bugsnag')
+  const objA = {};
+  objA.ref = objA;
+
+  // this tests the top level object being referred to as a node
+  expect(typedMap(objA)).toEqual({ ref: { type: 'string', value: '[circular]' } })
+
+  const objB = {};
+  objB.leafA = {};
+  objB.leafB = {};
+  objB.leafA.ref = objB.leafB
+  objB.leafB.ref = objB.leafA
+
+  // this tests subtrees referring to eachother
+  expect(typedMap(objB)).toEqual({
```

Comment:
Is the order in which the trees will be traversed guaranteed?  To avoid false negatives on this test.

Reply:
I don't think it's _guaranteed™_, but I'm almost certain that V8 will always traverse it like this.